pennylane/operation.py

```diff
@@ -964,6 +964,77 @@ def kraus_matrices(self):
         return self._kraus_matrices(*self.parameters)
 
 
+class Projection(Operation):
+    """Base class for projection operations
+
+    As with :class:`~.Operation`, the following class attributes must be
+    defined for all projections:
+
+    * :attr:`~.Operator.num_params`
+    * :attr:`~.Operator.num_wires`
+    * :attr:`~.Operator.par_domain`
+
+    To define a projection, the following attribute of :class:`~.Projection`
+    can be used to list the corresponding projectors.
+
+    * :attr:`~.Projection._projectors`
+
+    Args:
+        params (tuple[float, int, array, Variable]): operation parameters
+
+    Keyword Args:
+        wires (Sequence[int]): Subsystems the projection acts on. If not given, args[-1]
```

Comment:
```suggestion
        wires (Iterable[Number, str], Number, str, Wires): Subsystems the projection acts on. If not given, args[-1]
```
This is to support the `Wires` class